### PR TITLE
Getting a receipt requires waiting beyond just "OK", we want SUCCESS …

### PR DESCRIPTION
**Pull request by gregscullard**
…or an error

### CODE REVIEW

File: src/main/java/com/hedera/hashgraph/sdk/Transaction.java [hunk 1]
```diff
@@ -197,7 +197,11 @@ protected TransactionId mapResponse(TransactionResponse response) throws HederaE
                 }
             } else {
                 HederaException.throwIfExceptional(receiptStatus);
-                return mapReceipt.apply(receipt);
+                if (receiptStatus != ResponseCodeEnum.OK) {
```
Comment (mehcode) [inline src/main/java/com/hedera/hashgraph/sdk/Transaction.java]: This should probably be in the earlier `if` condition as that is what is going to wait for the retry delay. I also wouldn't do `!= OK` but `== OK` added to the condition (so errors still fall out).

Comment (gregscullard) [reply src/main/java/com/hedera/hashgraph/sdk/Transaction.java]: Made further updates/changes per your comments.

Comment (gregscullard) [reply src/main/java/com/hedera/hashgraph/sdk/Transaction.java]: Also reduced MAX_RETRY_ATTEMPT, 100 was unnecessary large
Also made the first wait > 0 (it was RETRY_DELAY * 0).

**Comment by gregscullard** [comment]
Superseded

**Pull request closed without merging**